Editor/Utility/CinemachineEditorAnalytics.cs

```diff
@@ -0,0 +1,257 @@
+        /// <param name="state">State change to detect entering playmode</param>
+        static void SendAnalyticsOnPlayEnter(PlayModeStateChange state)
+        {
+            // Only send analytics if it is enabled
+            if (!EditorAnalytics.enabled)
+                return;
+
+            // Only send data when entering playmode
+            if (state != PlayModeStateChange.EnteredPlayMode)
+                return;
+
+            var startTime = Time.realtimeSinceStartup;
+
+            var cinemachineCore = CinemachineCore.Instance;
+            var vcamCount = cinemachineCore.VirtualCameraCount;
+            var vcamDatas = new List<VcamData>();
+
+            // collect data from all vcams
+            for (int i = 0; i < vcamCount; ++i)
+            {
+                var vcamBase = cinemachineCore.GetVirtualCamera(i);
+                CollectData(vcamBase, i.ToString(), ref vcamDatas);
+            }
+
+            var projectData = new ProjectData
+            {
+                brain_count = cinemachineCore.BrainCount,
+                vcam_count = cinemachineCore.VirtualCameraCount,
+                cam_count = Camera.allCamerasCount,
+                vcams = vcamDatas,
+                time_elapsed = Time.realtimeSinceStartup - startTime,
+            };
+
+            // Register our event
+            EditorAnalytics.RegisterEventWithLimit("cm_vcams_on_play", 
+                k_MaxEventsPerHour, k_MaxNumberOfElements, k_VendorKey);
+
+            // Send the data to the database
+            EditorAnalytics.SendEventWithLimit("cm_vcams_on_play", projectData);
+        }
+
+        static void CollectData(CinemachineVirtualCameraBase vcamBase, string id, ref List<VcamData> vcamDatas)
+        {
+            if (vcamBase == null) return;
+
+            var vcam = vcamBase as CinemachineVirtualCamera;
+            if (vcam != null)
+            {
+                vcamDatas.Add(ConvertVcamToVcamData(vcam, id));
+            }
+            else // Composite vcam (Freelook, Mixing, Statedriven, Clearshot):
+            {
+                GetExtensions(vcamBase, out List<string> vcamExtensions, out int customExtensionCount);
+
+                vcamDatas.Add(new VcamData
+                {
+                    id = id,
+                    vcam_class = GetTypeName(vcamBase.GetType()),
+                    has_follow_target = vcamBase.Follow != null,
+                    has_lookat_target = vcamBase.LookAt != null,
+                    blend_hint = "",
+                    inherit_position = false,
+                    standby_update = vcamBase.m_StandbyUpdate.ToString(),
+                    mode_overwrite = "",
+                    body_component = "",
+                    aim_component = "",
+                    noise_component = "",
+                    custom_component_count = 0,
+                    extensions = vcamExtensions.ToArray(),
+                    custom_extension_count = customExtensionCount,
```

Comment:
All of this code (with the exception of body/aim/noise/customComponent) is repeated below.  It will get repeated again for NewVirtualCamera unless you clean a little more.  

Instance the structure once, and initialize the fields that are common for all vcamBase - which is most of them.  Then call custom code where applicable (vcam, newVcam) to fill in the body/aim/noise components.  Then add it to the list.

Reply:
Good points 👍